content/learning-paths/servers-and-cloud-computing/glibc-with-lse/mongo_benchmark.md

```diff
@@ -0,0 +1,140 @@
+operationcount=1000  
+workload=site.ycsb.workloads.CoreWorkload  
+readallfields=true  
+readproportion=0.2  
+updateproportion=0.3  
+scanproportion=0.3  
+insertproportion=0  
+readmodifywriteproportion=0.2  
+requestdistribution=zipfian  
+
+```
+
+## YCSB Run
+ 1. To run YCSB, you need following the `load` command first: 
+    ```console
+    ~/ycsb-0.17.0/bin/ycsb.sh load mongodb -s -P ~/ycsb-0.17.0/workloads/iworkload -p recordcount=10000000 -threads 256 -p mongodb.url="mongodb://${mongo_ip}:${mongo_port}/mymongodb"
+    ```
+    You can see the result after the `load` command finishes:
+    ```out
+    /usr/bin/java  -classpath /root/workload/tools/ycsb-0.17.0/conf:/root/workload/tools/ycsb-0.17.0/lib/core-0.17.0.jar:/root/workload/tools/ycsb-0.17.0/lib/HdrHistogram-2.1.4.jar:/root/workload/tools/ycsb-0.17.0/lib/htrace-core4-4.1.0-incubating.jar:/root/workload/tools/ycsb-0.17.0/lib/jackson-core-asl-1.9.4.jar:/root/workload/tools/ycsb-0.17.0/lib/jackson-mapper-asl-1.9.4.jar:/root/workload/tools/ycsb-0.17.0/mongodb-binding/lib/logback-classic-1.1.2.jar:/root/workload/tools/ycsb-0.17.0/mongodb-binding/lib/logback-core-1.1.2.jar:/root/workload/tools/ycsb-0.17.0/mongodb-binding/lib/mongodb-async-driver-2.0.1.jar:/root/workload/tools/ycsb-0.17.0/mongodb-binding/lib/mongodb-binding-0.17.0.jar:/root/workload/tools/ycsb-0.17.0/mongodb-binding/lib/mongo-java-driver-3.8.0.jar:/root/workload/tools/ycsb-0.17.0/mongodb-binding/lib/slf4j-api-1.7.25.jar:/root/workload/tools/ycsb-0.17.0/mongodb-binding/lib/snappy-java-1.1.7.1.jar site.ycsb.Client -load -db site.ycsb.db.MongoDbClient -s -P /root/workload/tools/ycsb-0.17.0/workloads/iworkloadf -p recordcount=10000000 -threads 256 -p mongodb.url=mongodb://172.26.202.189:27017/mymongodb
+    mongo client connection created with mongodb://172.26.202.189:27017/mymongodb
+    [OVERALL], RunTime(ms), 241978
+    [OVERALL], Throughput(ops/sec), 41326.07096512906
+    [TOTAL_GCS_G1_Young_Generation], Count, 334
+    [TOTAL_GC_TIME_G1_Young_Generation], Time(ms), 723
+    [TOTAL_GC_TIME_%_G1_Young_Generation], Time(%), 0.2987874930778831
+    [TOTAL_GCS_G1_Old_Generation], Count, 0
+    [TOTAL_GC_TIME_G1_Old_Generation], Time(ms), 0
+    [TOTAL_GC_TIME_%_G1_Old_Generation], Time(%), 0.0
+    [TOTAL_GCs], Count, 334
+    [TOTAL_GC_TIME], Time(ms), 723
+    [TOTAL_GC_TIME_%], Time(%), 0.2987874930778831
+    [CLEANUP], Operations, 256
+    [CLEANUP], AverageLatency(us), 14.421875
+    [CLEANUP], MinLatency(us), 0
+    [CLEANUP], MaxLatency(us), 3581
+    [CLEANUP], 95thPercentileLatency(us), 2
+    [CLEANUP], 99thPercentileLatency(us), 2
+    [INSERT], Operations, 10000000
+    [INSERT], AverageLatency(us), 6153.2624904
+    [INSERT], MinLatency(us), 183
+    [INSERT], MaxLatency(us), 969215
+    [INSERT], 95thPercentileLatency(us), 7563
+    [INSERT], 99thPercentileLatency(us), 14991
+    [INSERT], Return=OK, 10000000
+    ```
+
+ 2. Then you can benchmark the performance of MongoDB following the `run` command:
```

Comment:
As the title of the learning path indicates optimizing the performance of mongodb using glibc with LSE can you perhaps show running the same benchmark with glibc without LSE and highlighting the performance impact. Without it there's only raw output from the run command and doesn't show the reader how they get any extra benefit.

Reply:
I have added one page "Compare the results with LSE and NoLSE" as you wish.